chain/messagepool/selection.go

```diff
@@ -43,6 +43,7 @@ func (mp *MessagePool) SelectMessages(ctx context.Context, ts *types.TipSet, tq
 	mp.curTsLk.Lock()
 	defer mp.curTsLk.Unlock()
 
+	//TODO confirm if we can switch to RLock here for performance
```

Comment:
Yes, that should be safe

Reply:
thank you! Logged for this enhancement issue here: https://github.com/filecoin-project/lotus/issues/10597